tests/draft-future/dynamicRef.json

```diff
@@ -384,5 +384,62 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "after leaving a dynamic scope, it should not be used by a $dynamicRef",
+        "schema": {
+            "$id": "main",
```

Comment:
`$id` can't be relative if there is no retrieval URI.

Reply:
I strongly disagree with that assertion, but by convention we use absolute URIs in tests, so I pushed an update.